software/contrib/sequential_switch.py

```diff
@@ -0,0 +1,319 @@
+    not necessary for now
+    """
+    
+    def __init__(self, parent):
+        self.parent = parent
+        
+    def on_button1(self):
+        self.parent.active_screen = self.parent.switch_screen
+        self.parent.on_trigger()
+    
+    def draw(self):
+        oled.fill(0)
+        oled.show()
+
+class SwitchScreen:
+    """Default display: shows what output is currently active
+    """
+    
+    def __init__(self, parent):
+        self.parent = parent
+        
+    def on_button1(self):
+        # the button can be used to advance the output
+        self.parent.on_trigger()
+        
+    def draw(self):
+        oled.fill(0)
+        
+        # Show all 6 outputs as a string on 2 lines to mirror the panel
+        switches = ""
+        for i in range(2):
+            for j in range(3):
+                out_no = i*3 + j
+                if out_no < self.parent.num_outputs:
+                    # output is enabled; is it hot?
+                    if self.parent.current_output == out_no:
+                        switches = switches + " [*] "
+                    else:
+                        switches = switches + " [ ] "
+                else:
+                    # output is disabled; mark it with .
+                    switches = switches + "  .  "
+                    
+            switches = switches + "\n"
+        
+        oled.centre_text(switches)
+        
+        oled.show()
+
+class MenuScreen:
+    """Advanced menu options screen
+    """
+    def __init__(self, parent):
+        self.parent = parent
+        
+        self.menu_items = [
+            NumOutsChooser(parent),
+            ModeChooser(parent)
+        ]
+        
+    def draw(self):
+        self.menu_items[self.parent.menu_item].draw()
+        
+    def on_button1(self):
+        self.menu_items[self.parent.menu_item].on_button1()
+
+class NumOutsChooser:
+    """Used by MenuScreen to choose the number of outputs
+    """
+    
+    def __init__(self, parent):
+        self.parent = parent
+        
+    def read_num_outs(self):
+        return round(knob_rescale(k2, 2, 6)) # 2-6 outputs; 1 output would be useless!
+        
+    def on_button1(self):
+        num_outs = self.read_num_outs()
+        self.parent.num_outputs = num_outs
+        self.parent.current_output = self.parent.current_output % num_outs
+        self.parent.save()
+        
+    def draw(self):
+        num_outs = self.read_num_outs()
+        oled.fill(0)
+        oled.text(f"-- # Outputs --", 0, 0)
+        oled.text(f"{self.parent.num_outputs} <- {num_outs}", 0, 10)
+        oled.show()
+    
+class ModeChooser:
+    """Used by MenuScreen to choose the operating mode
+    """
+    def __init__(self, parent):
+        self.parent = parent
+        
+        self.mode_names = [
+            "Seq.",
+            "Rev.",
+            "P-P",
+            "Rand."
+        ]
+        
+    def read_mode(self):
+        return round(knob_rescale(k2, 0, len(self.mode_names)-1))
```

Comment:
[question] Why not use the built in range function here, as well as some of the other places you are using rescale?

Reply:
Because when I started writing it I got excited and started implementing stuff and didn't realize there was already a range function that did it all for me...?  And then I just got into the flow and kept re-using what I'd already written.